src/robot/running/bodyrunner.py

```diff
@@ -556,3 +565,81 @@ def _run_finally(self, data, run):
                 return err
             else:
                 return None
+
+
+class WhileLimitExceeded(RuntimeError):
+    pass
+
+
+class WhileLimit:
+
+    @staticmethod
+    def parse_iteration_count(limit):
+        try:
+            limit = int(limit)
+            if limit <= 0:
+                raise ValueError
+            return limit
+        except ValueError:
+            raise ValueError("Iteration limit must be positive integer when using "
+                             f"'x' or 'times', got: '{limit}'")
```

Comment:
This isn't part of the class API, right? Should be prefixed with `_` and be after the public `create`.